### PR TITLE
Feedback feature branch

### PR DESCRIPTION
**Pull request by AdamMc331**
All of this has been reviewed, so as long as the build still passes we will merge. Also going to smoke test this again, too.

### CODE REVIEW

File: Droidcon-Boston/app/src/main/java/com/mentalmachines/droidcon_boston/views/rating/RatingDialog.kt [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+        loadPreviousFeedback()
+
+        return view
+    }
+
+    private fun initializeViewModel() {
+        viewModel = ViewModelProviders.of(this).get(RatingViewModel::class.java)
+
+        // should not be able to get to this dialog w/o being logged in
+        val userId = AuthController.userId!!
```
[the remark this answers is not in the document]
Comment (CCorrado) [reply Droidcon-Boston/app/src/main/java/com/mentalmachines/droidcon_boston/views/rating/RatingDialog.kt]: yeah, just checked out this branch and launched it

[the remark this answers is not in the document]
Comment (CCorrado) [reply Droidcon-Boston/app/src/main/java/com/mentalmachines/droidcon_boston/views/rating/RatingDialog.kt]: I was able to log in without it crashing

[the remark this answers is not in the document]
Comment (AdamMc331) [reply Droidcon-Boston/app/src/main/java/com/mentalmachines/droidcon_boston/views/rating/RatingDialog.kt]: I'm gonna check again. I'm a little concerned about this.

**Comment by AdamMc331** [comment]
My error logging in:

```
E/AuthUI: A sign-in error occurred.
    com.firebase.ui.auth.FirebaseUiException: Code: 10, message: 10: 
        at com.firebase.ui.auth.data.remote.GoogleSignInHandler.onActivityResult(GoogleSignInHandler.java:110)
        at com.firebase.ui.auth.ui.idp.SingleSignInActivity.onActivityResult(SingleSignInActivity.java:128)
        at android.app.Activity.dispatchActivityResult(Activity.java:7462)
        at android.app.ActivityThread.deliverResults(ActivityThread.java:4391)
        at android.app.ActivityThread.handleSendResult(ActivityThread.java:4440)
        at android.app.servertransaction.ActivityResultItem.execute(ActivityResultItem.java:49)
        at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:108)
        at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:68)
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1816)
        at android.os.Handler.dispatchMessage(Handler.java:106)
        at android.os.Looper.loop(Looper.java:193)
        at android.app.ActivityThread.main(ActivityThread.java:6718)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:493)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:858)
E/AuthUI: A sign-in error occurred.
    com.firebase.ui.auth.FirebaseUiException: Code: 10, message: 10: 
        at com.firebase.ui.auth.data.remote.GoogleSignInHandler.onActivityResult(GoogleSignInHandler.java:110)
        at com.firebase.ui.auth.ui.idp.SingleSignInActivity.onActivityResult(SingleSignInActivity.java:128)
        at android.app.Activity.dispatchActivityResult(Activity.java:7462)
        at android.app.ActivityThread.deliverResults(ActivityThread.java:4391)
        at android.app.ActivityThread.handleSendResult(ActivityThread.java:4440)
        at android.app.servertransaction.ActivityResultItem.execute(ActivityResultItem.java:49)
        at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:108)
        at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:68)
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1816)
        at android.os.Handler.dispatchMessage(Handler.java:106)
        at android.os.Looper.loop(Looper.java:193)
        at android.app.ActivityThread.main(ActivityThread.java:6718)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:493)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:858)
```

**Comment by AdamMc331** [comment]
@kenyee Is it possible we need to add a new SHA to firebase? https://stackoverflow.com/a/50614809/3131147

**Comment by AdamMc331** [comment]
Looks like it was the SHA1 issue. I am going to smoke test this a little further but pending any issues I find, I'll merge tonight.

**Comment by AdamMc331** [comment]
The only change I want to make is to make the input textCapSentences so it allows people to submit like real feedback. Will push that in a sec.

**Pull request merged**